lib/problem_check/group_email_credentials.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+##
+# If group SMTP or IMAP has been configured, we want to make sure the
+# credentials are always valid otherwise emails will not be sending out
+# from group inboxes. This check is run as part of scheduled admin
+# problem checks, and if any credentials have issues they will show up on
+# the admin dashboard as a high priority issue.
+class ProblemCheck::GroupEmailCredentials < ProblemCheck
```

Comment:
Moved from `GroupEmailCredentialsCheck`, and augmented to return `Problem` objects instead of hashes.